tests/unit/test_factory.py

```diff
@@ -213,3 +215,54 @@ def test_get_project_not_found(url, mapping, instances, exc_str):
             url=url, service_mapping_update=mapping, custom_instances=instances
         )
     assert exc_str in str(ex.value)
+
+
+@pytest.mark.parametrize(
+    "instances_in_dict,result_instances",
+    [
+        ({}, set()),
+        ({"github.com": {"token": "abcd"}}, {GithubService(token="abcd")}),
+        ({"gitlab": {"token": "abcd"}}, {GitlabService(token="abcd")}),
+        ({"pagure": {"token": "abcd"}}, {PagureService(token="abcd")}),
+        (
+            {
+                "pagure": {
+                    "token": "abcd",
+                    "instance_url": "https://src.fedoraproject.org",
+                }
+            },
+            {PagureService(token="abcd", instance_url="https://src.fedoraproject.org")},
+        ),
+        (
+            {"github.com": {"token": "abcd"}, "gitlab": {"token": "abcd"}},
+            {GithubService(token="abcd"), GitlabService(token="abcd")},
+        ),
+        (
+            {"github.com": {"github_app_id": "abcd", "github_app_private_key": "efgh"}},
+            {GithubService(github_app_id="abcd", github_app_private_key="efgh")},
```

Comment:
This is a test for the github_app.

One downside is the need for private_key and not the path to the private_key. It is more a problem with `GithubService.__init__` method. We can probably allow both arguments in it. (We can create an issue for that and move the loading code from Packit here.)

Reply:
makes sense to create the issue

Reply:
Done: https://github.com/packit-service/ogr/issues/162

Reply:
Done: https://github.com/packit-service/ogr/issues/162